### PR TITLE
feat: add non-TTY download progress logs

### PR DESCRIPTION
**Pull request by Levi080513**
## Issue

NEU-202

## Summary

Add log-based model download progress for non-interactive environments so container and actor logs show long-running downloader activity. Keep interactive terminal behavior unchanged, and align the downloader CLI with the existing model download marker helper.

## Changes

- Add a downloader `ProgressReporter` with interval validation, TTY detection, size formatting, and completed/aborted terminal logs.
- Wrap HuggingFace downloads with non-TTY progress logging and scoped upstream HF progress-bar suppression when supported.
- Wrap Local downloads with non-TTY progress logging and compute planned copy bytes outside interactive terminals.
- Route `python -m neutree.downloader` through `download_with_markers(...)`.
- Add co-located Python unit tests for progress behavior, backend integration, review-fix edge cases, and CLI marker routing.

### Unit test

- [x] RED: new tests failed before implementation because `progress.py`, backend progress hooks, and CLI marker routing were missing.
- [x] RED after local review: new tests failed for pre-populated destination delta accounting, Local planned-copy byte totals, and HuggingFace progress-bar restoration.
- [x] RED after Copilot review: `test_log_progress_suppressed_after_stop_requested` failed before implementation because `_log_progress()` still emitted `Test download progress: 7 B downloaded` after `_stop` was set.
- [x] `PYTHONPATH=python python3 python/neutree/downloader/test_progress.py -v` — 9 tests passed.
- [x] `PYTHONPATH=python python3 python/neutree/downloader/test_huggingface.py -v` — 6 tests passed.
- [x] `PYTHONPATH=python python3 python/neutree/downloader/test_local.py -v` — 14 tests passed.
- [x] `PYTHONPATH=python python3 python/neutree/downloader/test_markers.py -v` — 3 tests passed.
- [x] Aggregate unittest runner for the four downloader modules — 32 tests passed.
- [x] `PYTHONPATH=python python3 -m compileall -q python/neutree/downloader`.
- [x] `git diff --check`.
- [x] Local CLI smoke with `NEUTREE_DL_BACKEND=local`, `NEUTREE_DL_PROGRESS_INTERVAL=1`, and `NEUTREE_VERIFY_SKIP=1` showed start/done markers and progress/completed logs.
- [ ] `PYTHONPATH=python python3 -m pytest python/neutree/downloader/ -v` — not run locally because `pytest` is not installed in this environment.

### DB test

- [x] Not required. This change does not touch DB schema, migrations, SQL, RLS, or persistence behavior.

### E2E test

- [x] Step 5 validated head `ddb2ada62e6eb573c52fd934151457e13df743bc` on the E2E lease `lease-NEU-202-step5-20260618142011`.
- [x] Runtime artifact: `neutree-runtime:v1.1.0-neu202.20260618`; serve artifact: `neutree-serve:v1.1.0-neu202.20260618`; engine artifact: `engine-vllm:v0.11.2-neutree202-ray2.53.0`; imported engine package `vllm v0.11.2-neutree202`.
- [x] C2722599 K8s endpoint log observation: endpoint `neu202-k8s-ep-06190026` reached `Running`; cluster `neu202-k8s-06190026`; `model-downloader` initContainer log evidence by marker type:
  - `NEUTREE_MODEL_DOWNLOAD_START`: 1 line.
  - `Local download progress:`: 26 lines, from `0 B / 953.3 MiB (0.0%)` through `918.0 MiB / 953.3 MiB (96.3%)`.
  - `Local download completed:`: 1 line, `953.3 MiB downloaded in 26.0s`.
  - `NEUTREE_MODEL_DOWNLOAD_DONE`: 1 line.
- [x] C2722600 Ray/SSH endpoint log observation: endpoint `neu202-ssh-ep-06190026` reached `Running`; cluster `neu202-ssh-06190026`; backend replica `int7ysn6` log evidence by stream:
  - stdout: `NEUTREE_MODEL_DOWNLOAD_START` 1 line and `NEUTREE_MODEL_DOWNLOAD_DONE` 1 line.
  - stderr: `Local download progress:` 14 lines, from `0 B / 953.3 MiB (0.0%)` through `866.1 MiB / 953.3 MiB (90.8%)`.
  - stderr: `Local download completed:` 2 lines, including `953.3 MiB downloaded in 6.4s`.
- [x] Code E2E not added because the approved NEU-202 design classifies this revision as manual/environment log observation. Unit tests cover reporter/backend/CLI contracts; manual Step 5 covers K8s and Ray/SSH runtime log visibility.
- [x] Additional HuggingFace SSH/vLLM live validation used engine package `vllm v0.11.2-neutree202` and image `engine-vllm:v0.11.2-neutree202-ray2.53.0`; endpoint `e2e-ep-ssh-chat-486558`; cluster `e2e-ep-ssh-486558`; model `Qwen/Qwen2.5-0.5B-Instruct:main`; model registry type `hugging-face`.
  - Ray Serve config confirmed `runtime_env.container.image=192.168.22.100/neutree-e2e-test/neutree/engine-vllm:v0.11.2-neutree202-ray2.53.0`, `model.registry_type=hugging-face`, and `model.registry_path=Qwen/Qwen2.5-0.5B-Instruct`.
  - stdout: `NEUTREE_MODEL_DOWNLOAD_START` and `NEUTREE_MODEL_DOWNLOAD_DONE`; backend also printed `[Backend] Model download completed.`
  - stderr: `HuggingFace download progress:` lines at `0 B`, `918.9 MiB`, and `953.3 MiB`; terminal log `HuggingFace download completed: 953.3 MiB downloaded in 172.0s`.
  - vLLM SSH HuggingFace chat specs passed for deploy/reach Running, normal inference, and wrong model-name error handling.
  - The broad label filter also matched an unrelated SGLang chat spec; it was stopped after the vLLM HuggingFace target specs passed, and the `486558` endpoint/cluster/modelregistry/imageregistry resources were cleaned up.
- [ ] A fresh engine rebuild from final PR head `d165502a8e6e65018e3086f661a9f58f1f054198` was not run before merge. The additional HuggingFace live validation above used the existing NEU-202 engine artifact built earlier for Step 5; current-head review fixes are covered by unit tests and compile checks.

## Risk & Rollback

No DB or API compatibility risk. Rollback is reverting this PR; existing downloader behavior will continue without periodic non-TTY progress logs and the CLI marker routing change.

### CODE REVIEW

File: python/neutree/downloader/progress.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    if size < 1024:
+        return f"{size} B"
+
+    value = float(size)
+    for unit in ("KiB", "MiB", "GiB", "TiB"):
+        value /= 1024.0
+        if value < 1024.0 or unit == "TiB":
+            return f"{value:.1f} {unit}"
+
+    return f"{value:.1f} TiB"
```
[the remark this answers is not in the document]
Comment (Levi080513) [reply python/neutree/downloader/progress.py]: Applied in d165502a8e6e65018e3086f661a9f58f1f054198. Removed the unreachable final return from `format_size()`; the TiB branch already returns inside the loop.

File: python/neutree/downloader/progress.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            return
+
+        if self.total_size:
+            pct = min(100.0, (size / self.total_size) * 100.0)
+            self.logger.info(
+                f"{self.label} progress: {format_size(size)} / "
+                f"{format_size(self.total_size)} ({pct:.1f}%)")
+            return
+
+        self.logger.info(f"{self.label} progress: {format_size(size)} downloaded")
```
[the remark this answers is not in the document]
Comment (Levi080513) [reply python/neutree/downloader/progress.py:157]: Applied in d165502a8e6e65018e3086f661a9f58f1f054198. `_log_progress()` now checks `_stop` before and after the destination-size scan, and suppresses debug logging if shutdown is requested during a scan exception. Added `test_log_progress_suppressed_after_stop_requested` to cover the shutdown race.

File: python/neutree/downloader/huggingface.py [hunk 3]
```diff
@@ -73,11 +110,20 @@ def download(self, source: str, dest: str, *, credentials: Optional[Dict[str, st
         # Empty string would be treated as an invalid branch name
         version = metadata.get("version") or None if metadata else None
```
[the remark this answers is not in the document]
Comment (Levi080513) [reply python/neutree/downloader/huggingface.py:111]: Not applied. Verified the Python precedence here: `metadata.get("version") or None if metadata else None` parses as `metadata ? (metadata.get("version") or None) : None`, so the guard is evaluated before `metadata.get(...)`. A runtime check with `metadata = None` returns `None` and does not raise `AttributeError`.

**Review by Yuyz0112 — APPROVED** [review]
